JBPM-6391: Refactoring shape views

romartin opened the pull request:
Hey @manstis @hasys 

Well first of all sorry for the big commit... had been working on improvements for some client side stuff and it were all related, cannot split. Appreciate if you can take a look asap as @wmedvede  and @alepintus depend on this <3

Here are the key points for this commit:

*  Improved SVG/CSS support 
** This way given as an input a set of SVG files and a single CSS one, it generates the shape view implementations (lienzo) by using the declared attributes in the style sheet declaration
** The idea is to obtain the same resulting shape in the canvas as the UX guys expect by providing us the SVG and CSS

* Removed the default bean property values realted to view attributes (like colors, borders, etc)
** By default beans do not provide values about the view - this way those are coming from the view declarations their selves (like the svg ones)
** On the other hand, when loading existing diagrams, if the generated beans contain information for that attributes, those are applied rather than the defaults from the views

* Refactored `MutableShapeDef` by `ShapeViewDef` (and all the shape definitions)
** The `MutableShapeDef` was providing an imperative declaration for shape attributes by using interface methods (`getAlpha`, `getColor`, etc), which is not much flexible to perform more advanced changes on the shapes...
** The new `ShapeViewDef` avoids the accidental complexity resulting on each shape def class (avoids implementing all methods even if you don't care about it...) and provides a decoupled way to change the views when a bean is updated, by using functional interfaces. See as example `EventCancelActivityViewHandler`.

* Re-styling for BPMN, CM and DMN shapes (UX will keep improving them  from this point)

PS: Noticed a bug when creating new shapes from the toolbox, they're not aligned with the previous one - @alepintus  please can you take it a look once merged? Thanks!

@wmedvede @alepintus : Sorry for the delay on this commit, was not easy.. anyway hope you can rebase from it when merged and be able to move forward.

Thanks!

romartin commented:
Jenkins please retest this

romartin commented:
Jenkins please retest this

romartin commented:
Jenkins please retest this

hasys commented:
Jenkins execute full downstream build

hasys commented:
Jenkins execute full downstream build

hasys commented:
Jenkins execute full downstream build

manstis commented:
Jenkins please retest this.

manstis commented:
@romartin I think you'll need to rebase this with ```master``` to compile successfully now.

romartin commented on kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-client-common/src/main/java/org/kie/workbench/common/stunner/core/client/shape/impl/ShapeStateStrokeHandler.java:
```diff
@@ -0,0 +1,223 @@
+                runState(ShapeState.INVALID);
+            } else {
+                runStrokeStateHandler(strokeState);
+            }
```
yep agree, good point

romartin commented on kie-wb-common-stunner/kie-wb-common-stunner-extensions/kie-wb-common-stunner-svg/kie-wb-common-stunner-svg-gen/src/main/java/org/kie/workbench/common/stunner/svg/gen/codegen/impl/SVGPrimitiveGeneratorUtils.java:
```diff
@@ -0,0 +1,103 @@
+        try {
+            final StringBuffer childBuffer = generatorProvider.apply(child).generate(child);
+            final boolean scalable = child.isScalable();
+            final String scalableRaw = scalable ? "true" : "false";
```
agree too! it seems I need to refresh some Java again...  :)

romartin commented:
Jenkins please retest this

romartin commented:
thanks @hasys @manstis ! Was not an easy one... I've applied the changes proposed and rebased, once all green I'll merge, as several people depends on this... 
thanks!!

romartin commented:
Jenkins please retest this

romartin commented:
Jenkins please retest this

hasys commented:
Jenkins execute full downstream build

romartin commented:
Hey @hasys 

Great job! Thanks for testing!

Had been trying to fix as much as possible, added a new commit, here are the comments:

1.- **Memory leak** - Tbh not see the difference between this commit and the current upstream code, I have both versions running here both are... leaking similar :/ Had been doing a quick look to see if some fix sounds easy to do but no luck.. so I moved [this ticket](https://issues.jboss.org/browse/JBPM-5462) to the top of the backlog, in order to give it priority asap, we must fix this urgently, but difficult to achieve in a few hours on top of this.
2.- **Wrong node location** (using toolbox buttons) - Had been fixing it too!! was not easy and it's still not perfect.. but looks better as before, hope that's fine to move forward. Anyway just opened [this new bug](https://issues.jboss.org/browse/JBPM-6629) and I'll work with Alessio on it during this week.
3.- Agree, in fact I'm not going to close the [JIRA ticket](https://issues.jboss.org/browse/JBPM-6391) for this task, the idea is wait for UXD improvements from that point, will reassign it, but this commit is the one that allows them to easily do it (you'd been on latest meetings about this)
4.- Fixed
5.- Yeah let's open a new issue for this! :+1: 

Thanks!
Roger

hasys commented:
Jenkins execute full downstream build

hasys commented:
Jenkins execute full downstream build

hasys commented:
Hi @romartin, I am not able to test kie-wb due to some GWT compilation errors, sorry for delay.

hasys commented:
Jenkins execute full downstream build

romartin commented:
Jenkins please retest this

romartin commented:
Jenkins please retest this

romartin commented:
ohh @hasys !!! It's green again! need your feedback for being able to merge this... <3

romartin commented on kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-client-common/src/main/java/org/kie/workbench/common/stunner/core/client/canvas/util/CanvasLayoutUtils.java:
```diff
@@ -41,22 +42,6 @@
- * <p>
- * TODO: This has to be refactored by the use of a good impl that achieve good dynamic layouts. Probably each
- * Definition Set / Diagram will require a different layout manager as well.
- */
```
yeah that JavaDoc  is no longer up-to-date... so better removing it. Alessio had been working here and this has changed a lot since the beginning...

romartin commented:
great, thanks @hasys !!
